### PR TITLE
tests for Promise.try (stage 1)

### PR DESCRIPTION
**Pull request by dnalborczyk**
tests for Promise.try, which is currently "stage 1" in the TC39 process.

https://github.com/tc39/proposals
https://github.com/ljharb/proposal-promise-try

### CODE REVIEW

**Comment by zloirock** [comment]
сс @ljharb

File: data-esnext.js [hunk 1]
```diff
@@ -1794,6 +1794,112 @@ exports.tests = [
...
+{
+  name: 'Promise.try',
+  category: STAGE1,
+  significance: 'small',
+  spec: 'https://github.com/ljharb/proposal-promise-try',
+  'subtests': [
+    {
+      name: 'basic support',
+      exec: function () {/*
+        return typeof Promise.try !== 'undefined';
```
Comment (ljharb) [inline data-esnext.js]: This should be checking that it's a function, instead of just "not undefined"

**Review by ljharb — CHANGES_REQUESTED** [review]
Thanks, this is great!

Can we also add tests that ensure it throws if it lacks a receiver (`this` value); that it works with thenables returned from the function' and that it works with subclassing?

**Comment by ljharb** [comment]
Can we also make assertions that it has the name "try" and that it has a length of 1?

**Pull request closed without merging**